curated_transformers/models/feedforward.py

```diff
@@ -0,0 +1,63 @@
+import torch
+from torch import Tensor
+from torch.nn import Linear, Module
+
+from .activations import GeluNew
+from ..errors import Errors
+
+
+class PointwiseFeedForward(Module):
+    """Point-wise feed-forward layer (Vaswani et al., 2017).
+
+    This layer is applied pointwise, meaning that the same
+    transformation is applied to each sequence element. This
+    transformation is: g(xW_1 + b_1)W_2 + b_2, where W_1 and b_1 transform
+    the input to an intermediate width, g is a non-linear activation
+    function and W_2 and b_2 transform the output of the activation back to
+    the input width.
+
+    hidden_act (str): the activation function to apply, one of: "relu",
+        "gelu" or "gelu_new" (default: "gelu").
+    hidden_width (int): the input and output width of the layer.
+        (default: 768)
+    intermediate_width (int): the width of the projection to which the
+        non-linearity is applied. (default: 3072)
+    """
+    def __init__(
+        self,
+        *,
+        hidden_act: str = "gelu",
+        hidden_width: int = 768,
+        intermediate_width: int = 3072,
+    ):
+        super().__init__()
+
+        self.intermediate = Linear(hidden_width, intermediate_width)
+        self.output = Linear(intermediate_width, hidden_width)
+        if hidden_act == "relu":
+            self.activation = torch.nn.ReLU()  # type: ignore
+        elif hidden_act == "gelu":
+            self.activation = torch.nn.GELU()  # type: ignore
+        elif hidden_act == "gelu_new":
+            # Ideally, we would use torch.nn.GELU(approximate="tanh"). However,
+            # the differences between that and the manual Torch implementation
+            # are large enough to fail tests comparing output to HF
+            # transformers.
+            self.activation = GeluNew()  # type: ignore
+        else:
+            raise ValueError(
+                Errors.E004.format(activation_funcs=("relu", "gelu", "gelu_new"))
+            )
```

Comment:
It works here, because the error is in the constructor and TorchScript does not need to capture constructors. This is also covered by the TorchScript conversion test, since the factored out layers is used in the BERT encoder and we do test TorchScript conversion of the BERT encoder.

Reply:
I agree with the inlining. It would be weird to have both styles. I'll create a ticket for that.